tests/behat/features/capabilities/book/book-create.feature

```diff
@@ -14,3 +14,16 @@ Feature: Create Book page
 
     Then I should see the book I just created
     And it should not show author information
+
+  Scenario: User can create "book" node as book by default
+    Given I am logged in as an "administrator"
+
+    When I am on "node/add/book"
+    Then I should see "- Create a new book -" in the "#edit-book-bid" element
```

Comment:
I think there's two ways we can improve this test depending on what we want to test exactly:

1. We probably mean to check that "A book can not be outside of a book structure" which would be a slightly different test
2. We find it important to ensure that the default is a new book in which case we could rephrase the scenario slightly "The book content type create a new book structure by default". We should still add coverage for scenario 1 in a separate test, since that's part of business logic added in this PR.

In either case we should avoid testing as administrator because it bypasses all access checks.

Suggestion for scenario 1:
```suggestion
  Scenario: A book type node can not be outside of a book structure
    Given I am logged in as a user with the verified role

    When I view the book creation page

    Then I should not see "- None -" in the Book select field
```

I've changed a few things:
1. Changed the scenario name to reflect the added business logic
2. Test as the least privileged role that should be able to do this
3. Remove the URL from the test to make it more human readable, this requires a new function in `BookContext` (in a different PR I'm introducing similar steps for topics and events)
4. Remove the use of a field ID and instead rely on a human readable label of the field. This helps the test also test accessibility and makes it easier to manually verify the test. This requires adding a custom step for `SocialMinkContext`
5. Added whitespace between "When" and "Then" so that the [arrange (Given), act (When), assert (Then)](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) structure is clearer.

The custom step for 3 would be 
```php
  /**
   * View the book creation page.
   *
   * @When /^(?:|I )view the book creation page$/
   */
  public function whenIViewTheBookCreationPage() : void {
    $this->visitPath(self::CREATE_PAGE);
    $this->assertSession()->statusCodeEquals(200);
  }
 ```
  
 The custom step for 4 would be 
```php 
  /**
   * Ensure a select field does not contain an option.
   *
   * @Then I should not see :option in the :locator select field
   * @Then should not see :option in the :locator select field
   */
  public function thenShouldNotSeeOptionInTheSelectField(string $option, string $locator) : void {
    $field = $this->getSession()->getPage()->findField($locator);

    if (null === $field) {
      throw new ElementNotFoundException($this->getSession()->getDriver(), 'form field', 'id|name|label|value', $locator);
    }

    $opt = $field->find('named', ['option' => $option]);
    
    if ($opt !== NULL) {
      throw new \Exception("The field was not supposed to contain '$option' but it was an option in the select field.");
    }
  }
 ```

If we also want to test that "- Create a new book -" is the default we could keep the current test in addition to the new one for disallowing `- None -` and modify it slightly:
```suggestion
  Scenario: A new book is created by default
    Given I am logged in as an "verified"

    When I view the book creation page

    Then I should see "- Create a new book -" selected in the Book select field
```

I've made similar changes as I've done for the other scenario. We're also adding another step for our select fields to `SocialMinkContext` which relies on field labels rather than IDs.

The new step can be implemented as 
```php
  /**
   * Ensure a specific option is selected in a select field.
   *
   * @Then I should see :option selected in the :locator select field
   * @Then should see :option selected in the :locator select field
   */
  public function thenShouldSeeOptionSelected(string $option, string $locator) : void {
    $field = $this->getSession()->getPage()->findField($locator);

    if (null === $field) {
      throw new ElementNotFoundException($this->getSession()->getDriver(), 'form field', 'id|name|label|value', $locator);
    }

    $opt = $field->find('named', ['option', $option]);

    if (NULL === $opt) {
      throw new ElementNotFoundException($this->getDriver(), 'select option', 'value|text', $option);
    }

    if (!$opt->isSelected()) {
      throw new \Exception("Expected '$option' to be selected but it was not.");
    }
  }
```

Reply:
Added